scripts/build-tools.sh

```diff
@@ -39,13 +39,21 @@ make_tool()
         # if no argument provided, all the tools will be built. Empty tool is
         # okay.
         tool=$1
-        # shellcheck disable=SC2086
-        make -C "$BUILD_TOOLS_DIR" -j "$NO_PROCESSORS" $tool
+
+        # 'cd' is unfortunately much less verbose than 'make -C'.
+        # 'MAKEFLAGS=--no-print-directory' works too but let's not
+        # interfere with the user's environment.
+        ( cd "$BUILD_TOOLS_DIR"
```

Comment:
does this return to the previous directory after? Otherwise pushd and popd might be worth using

Reply:
Yes it does because `(  )` is a subshell / subprocess so the main script doesn't even know that something changed directory, no side-effect on the parent. This is in general better than `pushd` because the subshell can exit (voluntarily or not) at any point without worrying about restoring anything.

`pushd` is of course more convenient in interactive use or when you actually need some side effects.

For (a lot!) more on these topics this is a great resource: http://mywiki.wooledge.org/BashFAQ/024